src/solver/hydro/daily/h2o_j_lisser_les_sur_turbines.cpp

```diff
@@ -32,55 +32,43 @@
 
 void H2O_J_LisserLesSurTurbines(DONNEES_MENSUELLES* DonneesMensuelles, int NumeroDeProbleme)
 {
-    int Pdt;
-    int Cnt;
-    int NbPdt;
-    double X;
-    int Np;
-    char* Flag;
-    char LimiteAtteinte;
-    int NbCycles;
-    double SurTurbineARepartir;
-    double Xmoy;
-    double MargeMin;
-    double SurTurbine;
     PROBLEME_HYDRAULIQUE& ProblemeHydraulique = DonneesMensuelles->ProblemeHydraulique;
 
     std::vector<double>& TurbineMax = DonneesMensuelles->TurbineMax;
     std::vector<double>& Turbine = DonneesMensuelles->Turbine;
     std::vector<double>& TurbineCible = DonneesMensuelles->TurbineCible;
 
-    Flag = (char*) ProblemeHydraulique.ProblemeLineairePartieVariable[NumeroDeProbleme].Xmax.data();
+    char* Flag = (char*) ProblemeHydraulique.ProblemeLineairePartieVariable[NumeroDeProbleme].Xmax.data();
```

Comment:
I think this part is very error-prone
- We have a `std::vector<double>`
- We cast it's pointer to `char*`
- We use it as a temporary `Flag` object

The resulting `Xmax` contains essentially garbage. If we try to read it, we'll get inconsistent values such as 1.22e308.

If `Flag` is used only here, I strongly suggest declaring a `std::vector<bool>` here instead.